hydra-task/src/main/java/com/addthis/hydra/task/output/tree/LeasedTreeNodeList.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.addthis.hydra.task.output.tree;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import com.addthis.hydra.data.tree.DataTreeNode;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LeasedTreeNodeList extends ReadOnceList<DataTreeNode> {
+
+    static final Logger log = LoggerFactory.getLogger(LeasedTreeNodeList.class);
+
+    public static LeasedTreeNodeList unmodifiableList() {
+        return new LeasedTreeNodeList(Collections.unmodifiableList(new ArrayList<>()));
+    }
+
+    public LeasedTreeNodeList() {
+        super();
+    }
+
+    public LeasedTreeNodeList(int capacity) {
+        super(capacity);
+    }
+
+    protected LeasedTreeNodeList(List<DataTreeNode> nodes) {
+        super(nodes);
+    }
+
+    @Override protected void releaseItem(DataTreeNode data) {
+        data.release();
+    }
+
+    /**
+     * Should we delete this method is it worthless?
```

Comment:
Yes, we should delete it. It isn't worthless, but its cost far exceeds our its value. Creating possibly hundreds of thousands of finalizers a second is really, really not where we want to be.